cluster/src/main/java/com/hartwig/pipeline/tertiary/lilac/Lilac.java

```diff
@@ -29,27 +27,32 @@
 import com.hartwig.pipeline.report.RunLogComponent;
 import com.hartwig.pipeline.resource.ResourceFiles;
 import com.hartwig.pipeline.stages.Namespace;
+import com.hartwig.pipeline.stages.Stage;
 import com.hartwig.pipeline.storage.GoogleStorageLocation;
 import com.hartwig.pipeline.storage.RuntimeBucket;
-import com.hartwig.pipeline.tertiary.TertiaryStage;
 import com.hartwig.pipeline.tertiary.purple.PurpleOutput;
 import com.hartwig.pipeline.tertiary.purple.PurpleOutputLocations;
 import com.hartwig.pipeline.tools.Versions;
 
 @Namespace(Lilac.NAMESPACE)
-public class Lilac extends TertiaryStage<LilacOutput> {
+public class Lilac implements Stage<LilacOutput, SomaticRunMetadata> {
     public static final String NAMESPACE = "lilac";
 
     private final ResourceFiles resourceFiles;
+    private final LilacBamSliceOutput slicedOutput;
     private final InputDownload purpleGeneCopyNumber;
     private final InputDownload purpleSomaticVariants;
+    private final InputDownload referenceBam;
+    private final InputDownload tumorBam;
 
-    public Lilac(final AlignmentPair alignmentPair, final ResourceFiles resourceFiles, final PurpleOutput purpleOutput) {
-        super(alignmentPair);
+    public Lilac(final LilacBamSliceOutput slicedOutput, final ResourceFiles resourceFiles, final PurpleOutput purpleOutput) {
         this.resourceFiles = resourceFiles;
         PurpleOutputLocations purpleOutputLocations = purpleOutput.outputLocations();
         this.purpleGeneCopyNumber = initialiseOptionalLocation(purpleOutputLocations.geneCopyNumber());
         this.purpleSomaticVariants = initialiseOptionalLocation(purpleOutputLocations.somaticVariants());
+        this.slicedOutput = slicedOutput;
+        this.referenceBam = initialiseOptionalLocation(slicedOutput.reference());
+        this.tumorBam = initialiseOptionalLocation(slicedOutput.tumor());
```

Comment:
do we still need both the sliced and non-sliced bams? I think part of the idea here was to make this stage not depend on anything but the slice bam

Reply:
Yeah this is just a naming thing, I'm caching the paths to the sliced BAMs as `referenceBam` and `tumorBam`, and keeping a reference to the full `slicedOutput` to access the indices (which I only need in one place). Could just make a member variable for the indices too but felt like it was polluting the namespace a bit.